plugins/modules/discovery_intent.py

```diff
@@ -721,6 +721,43 @@ def validate_input(self, state=None):
         self.status = "success"
         return self
 
+    def validate_ip_address_list(self):
+        """
+        Validates each ip adress paased in the IP_address_list passed by the user before preprocessing it
+        """
+
+        ip_address_list = self.validated_config[0].get('ip_address_list')
+        for ip in ip_address_list:
+            if '/' in ip:
+                ip = ip.split("/")[0]
+            if '-' in ip:
+                if len(ip.split('-')) == 2:
+                    ip1, ip2 = ip.split('-')
+                    if self.is_valid_ip(ip1) is False:
+                        msg = "IP address {0} is not valid".format(ip1)
+                        self.log(msg, "CRITICAL")
+                        self.module.fail_json(msg=msg)
+                    if self.is_valid_ip(ip2) is False:
+                        msg = "IP address {0} is not valid".format(ip2)
+                        self.log(msg, "CRITICAL")
+                        self.module.fail_json(msg=msg)
+                    ip1_parts = list(map(int, ip1.split('.')))
+                    ip2_parts = list(map(int, ip2.split('.')))
+                    for part in range(4):
+                        if ip1_parts[part] > ip2_parts[part]:
+                            msg = "Incorrect range passed. Please pass correct IP address range"
+                            self.log(msg, "CRITICAL")
+                            self.module.fail_json(msg=msg)
+                else:
+                    msg = "IP address range should have only upper and lower limit values"
```

Comment:
What is the range? Can we print it?
msg = "Provided range '{0}' is incorrect. IP address range should have only upper and lower limit values".format(provided_range)